README.md

```diff
@@ -1 +1,77 @@
+
+The general architecture of the model (without adversarial examples) is pictured below. More details can be found within the Tf-Ranking Tutorial mentioned previously.
+
+![alt text](https://github.com/googleinterns/hotels-ranking-adversarial/blob/code-review/images/model_achitecture.JPG  "Model Architecture Diagram")
+
+For the purpose of generating adversarial examples, we focus on the model during the serving stage. The model takes in the raw textual data via the input_fn function and then converts it to numerical embeddings via the transform_fn function. This data can then be passed into to the scoring function and receive a corresponding ranking. When calculating the scores for unperturbed input the model simultaneously calculates the gradient required to generate adversarial noise. Then, the model can take in a perturbed input that consists of the answer embeddings in addition to our adversarial noise calculated previously.
+
+## Setup
+To run the files the following is required:
+1) Python with pip (compatible with versions 3.5 - 3.8) 
+  * Can be downloaded here: https://www.python.org/
+2) Tensorflow
+  * Can be downloaded here: https://www.tensorflow.org/install/pip
+3) Tensorflow Ranking
+  * Can be downloaded here: https://github.com/tensorflow/ranking
+4) Matplotlib
+  * Use pip to install library
+```
+pip install matplotlib
+```
+
+5) Dataset Files
+  * Use wget to download dataset. 
+```
+wget -O "vocab.txt" "http://ciir.cs.umass.edu/downloads/Antique/tf-ranking/vocab.txt"
+wget -O "train.tfrecords" "http://ciir.cs.umass.edu/downloads/Antique/tf-ranking/ELWC/train.tfrecords"
+wget -O "test.tfrecords" "http://ciir.cs.umass.edu/downloads/Antique/tf-ranking//ELWC/test.tfrecords"
+```
+## Running the Code
+The code can be executed by calling the following:
+```
+python main.py --directory=directory_name
+```
+If no argument is filled in for the directory, visualizations will be saved in the current directory by default. 
+
+Upon running the code, a query of will be displayed alongside a list of questions ranked in order of relevance. The user can then select a question to be perturbed, a reference answer, and the amount of perturbation. The reference answer serves to indicate the direction of the perturbation. For instance, if the reference answer has a higher rank than the specified answer to be perturbed, the perturbation will increase the rank of the specified answer. Then the model will rank the answers again, this time with the perturbed input. For comparison purposes, the model will rank the answers one last time with randomly generated (as opposed to generated with FGSM) noise to demonstrate the larger effect FGSM generated noise tends to have on the model. Then a bar plot demonstrating the rankings of both the unperturbed, FGSM-perturbed, and randomly perturbed inputs will be displayed and saved to the specified folder. Other visualizations (described below) can also be added.
+
+
+## Visualizations
```

Comment:
Add example figures? Make sure labels and legends are correct in example figures.